sys/net/gnrc/netreg/gnrc_netreg.c

```diff
@@ -38,15 +39,18 @@ void gnrc_netreg_init(void)
 
 int gnrc_netreg_register(gnrc_nettype_t type, gnrc_netreg_entry_t *entry)
 {
-#if defined(MODULE_GNRC_NETAPI_MBOX) || defined(MODULE_GNRC_NETAPI_CALLBACKS)
 #ifdef DEVELHELP
-    /* only threads with a message queue are allowed to register at gnrc */
-    assert((entry->type != GNRC_NETREG_TYPE_DEFAULT) ||
-           sched_threads[entry->target.pid]->msg_array);
+    bool has_msg_q = sched_threads[entry->target.pid]->msg_array;
+
+#if defined(MODULE_GNRC_NETAPI_MBOX) || defined(MODULE_GNRC_NETAPI_CALLBACKS)
+    has_msg_q = has_msg_q || (entry->type != GNRC_NETREG_TYPE_DEFAULT);
 #endif
-#else
     /* only threads with a message queue are allowed to register at gnrc */
-    assert(sched_threads[entry->target.pid]->msg_array);
+    if (!has_msg_q) {
+        LOG_ERROR("\n!!!! gnrc_netreg: initialize message queue of thread %u "
```

Comment:
Logging is supposed to be used for runtime errors. I know this is a boundary case, but this error points to a mistake in code.

Using LOG_ERROR here will enlarge every binary that does not disable logging (ERROR is the lowest level) by the size of the error message.

Shouldn't the function return an error in this case anyways?

Reply:
Ping @kaspar030!

Reply:
Now that even @aabadie [reports failed assertions as kernel panic](https://github.com/RIOT-OS/RIOT/pull/7898#issuecomment-344202573), maybe this PR gets some new traction ;-P